Fixed #36500 -- Set flake8 max-doc-length.

medmunds opened the pull request:
(This is "option 4" from https://forum.djangoproject.com/t/enforce-or-eliminate-comment-docstring-line-length-limit/41760. It would need additional manual edits before merging.)

#### Trac ticket number

ticket-36500

#### Branch description
Rewrapped long docstrings and block comments to 79 characters to match coding-style requirements.

(This also includes a no-ticket patch for a clearly incorrect attempt to concatenate a string in a test case, which was detected by adding flake8 max-doc-length but is unrelated to this change.)

#### Checklist
- [x] This PR targets the `main` branch. <!-- Backports will be evaluated and done by mergers, when necessary. -->
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" ticket flag in the Trac system.
- n/a I have added or updated relevant tests.
- [x] I have added or updated relevant docs, including release notes if applicable.
- n/a I have attached screenshots in both light and dark modes for any UI changes.

medmunds commented:
This pull request will quickly become outdated. To rerun the auto fixer (in the third commit):

- Grab the latest django/main
- Copy the [autofix_w505.py](https://github.com/medmunds/autofix-w505) script into the repo root
- Run `python autofix_w505.py .`

blighj reviewed (commented):
I did a review of the django folder so close to half the files. Took a good hour, even reviewing my 37 suggestions will be a commitment for someone. 

There are plenty of places where we've changed a not very long line of text (but one that is indented a few levels) and broke it up so that there is now one word on the end of a line. I think that is worse and comes from having a hard and fast rule.
To me the 80 char rule comes from a combination of terminals used to be 80 characters and readable paragraphs should be 50–75 chars. By setting a line limit of 88 we are already saying that the first reason is not as important any more. Even having done the review work on this I still favour a guidance for 50-75 for text with a hard limit at 88 rather than enforcing 79.

medmunds commented:
@blighj @nessita I should clarify I *don't* think this PR is ready for review yet. (That's why it's a draft.) Someone should do the manual reformatting *first* and then it should be reviewed.

That someone could be me (let me know), or it could be one of you (you are welcome to take over ownership of this PR). 

I'd suggest the manual reformatting should be limited to preserving the intent of the original formatting: fixing tables, fixing indentation in parameter lists, making sure code examples are valid code, etc. I'd like to think any of this manual reformatting could be included in git-blame-ignore along with the auto-reformatting, so every change should be something we'd be comfortable hiding from the history.

(Much as I'd be tempted to correct typos, add missing punctuation, reword to improve wrapping, etc., I think those should be separate from enforcing max-doc-length. Just my opinion.)

Incidentally, this PR is already stale and needs to be rebased thanks to #19663. (Which, ironically, was discovered due to this PR.)

medmunds commented on django/conf/global_settings.py:
```diff
@@ -22,7 +22,8 @@ def gettext_noop(s):
 
 # People who get code error notifications. In the format
-# [('Full Name', 'email@example.com'), ('Full Name', 'anotheremail@example.com')]
+# [('Full Name', 'email@example.com'), ('Full Name',
+# 'anotheremail@example.com')]
```
Outdated, needs rebase

medmunds commented on django/contrib/auth/password_validation.py:
```diff
@@ -227,7 +227,8 @@ class CommonPasswordValidator:
     The password is rejected if it occurs in a provided list of passwords,
     which may be gzipped. The list Django ships with contains 20000 common
-    passwords (unhexed, lowercased and deduplicated), created by Royce Williams:
+    passwords (unhexed, lowercased and deduplicated), created by Royce
+    Williams:
```
This is semantic line breaking. I'm generally in favor of that. I'm not sure it's appropriate as part of this "automated reformatting" PR.

nessita replied:
Let's please avoid changes like this. In general, the existing pattern is to use as much of the 79 columns as possible. I know this is not your preference, but in my opinion, it's easier for contributors of any background. The "semantic" meaning of a line break is much fuzzier for people who did not grow up with the language.

(Another example I can think of is books: they do not use semantic line wrapping, and they are quite readable, I think! :books: )

medmunds commented on django/db/backends/mysql/introspection.py:
```diff
@@ -195,8 +196,8 @@ def get_sequences(self, cursor, table_name, table_fields=()):
         """
-        Return a dictionary of {field_name: (field_name_other_table, other_table)}
-        representing all foreign keys in the given table.
+        Return a dictionary of {field_name: (field_name_other_table,
+        other_table)} representing all foreign keys in the given table.
```
I would probably also indent the code line (which prevents it from becoming filled if someone runs the script again):

```text
        Return a dictionary of
            {field_name: (field_name_other_table, other_table)}
        representing all foreign keys in the given table.
```

(An alternative would be putting it in backticks or double backticks to clarify that it's "code" and allowing it to wrap.)

medmunds replied:
Ditto

nessita commented on django/contrib/humanize/templatetags/humanize.py:
```diff
@@ -39,11 +39,14 @@ def ordinal(value):
             # Translators: Ordinal format when value ends with 0, e.g. 80th.
             pgettext("ordinal 0", "{}th"),
-            # Translators: Ordinal format when value ends with 1, e.g. 81st, except 11.
+            # Translators: Ordinal format when value ends with 1, e.g. 81st,
+            # except 11.
```
Let's not incorporate changes like this. The content should fit/use as much as possible of the 79cols.

nessita commented on django/contrib/humanize/templatetags/humanize.py:
```diff
@@ -39,11 +39,14 @@ def ordinal(value):
+            # except 11.
             pgettext("ordinal 1", "{}st"),
-            # Translators: Ordinal format when value ends with 2, e.g. 82nd, except 12.
+            # Translators: Ordinal format when value ends with 2, e.g. 82nd,
+            # except 12.
```
Ditto.

nessita commented on django/contrib/humanize/templatetags/humanize.py:
```diff
@@ -39,11 +39,14 @@ def ordinal(value):
+            # except 12.
             pgettext("ordinal 2", "{}nd"),
-            # Translators: Ordinal format when value ends with 3, e.g. 83rd, except 13.
+            # Translators: Ordinal format when value ends with 3, e.g. 83rd,
+            # except 13.
```
Ditto.

nessita commented on django/core/handlers/base.py:
```diff
@@ -27,7 +27,8 @@ def load_middleware(self, is_async=False):
         Populate middleware lists from settings.MIDDLEWARE.
 
-        Must be called after the environment is fixed (see __call__ in subclasses).
+        Must be called after the environment is fixed (see __call__ in
+        subclasses).
```
Ditto. (I will stop now, you get the point.)

medmunds commented on django/db/migrations/loader.py:
```diff
@@ -310,9 +314,8 @@ def build_graph(self):
-            # applied before. In that case raise a more understandable exception
-            # (#23556).
-            # Get reverse replacements.
+            # applied before. In that case raise a more understandable
+            # exception (#23556). Get reverse replacements.
```
I *think* this is logically two paragraphs, so the proposed manual change is correct

medmunds commented:
(preparing updates…)

medmunds commented:
OK, I've rebased and split this into four commits:
1. A pre-fix that applies my editorial judgment on wording/formatting to several things the autofixer would mangle.
2. The autofixer. I've taken the liberty of attributing this commit to django-bot (as was done with black)
3. Manual fixups for things the autofixer got wrong.
4. Changing flake8's max-doc-length.

I'd expect commits 2 and 3 could be git-blame-ignored. (The difference between 1 and 3 is that 1 made substantive changes that probably shouldn't be ignored.)

I'll drop a few comments inline as well.

medmunds commented on django/contrib/gis/gdal/srs.py:
```diff
@@ -116,7 +116,11 @@ def __getitem__(self, target):
+        ...     'GEOGCS["WGS 84",'
+        ...     '  DATUM["WGS_1984, ... AUTHORITY["EPSG","4326"]'
+        ...     ']'
+        ... )
```
I'm not thrilled with this editorial change, but it's equivalent to the original. (Given the syntax error in the original, I suspect it's psuedo-code anyway.)

medmunds commented on django/contrib/gis/utils/layermapping.py:
```diff
@@ -339,7 +340,7 @@ def check_unique(self, unique):
             )
 
-    # Keyword argument retrieval routines ####
+    # #### Keyword argument retrieval routines
```
Changed for consistency with other similar comments that were also getting edited

medmunds commented on django/core/management/commands/makemigrations.py:
```diff
@@ -505,9 +507,9 @@ def all_items_equal(seq):
-                    # to the disk.
+                    # Alternatively, makemigrations --merge --dry-run
+                    # --verbosity 3 will log the merge migrations rather than
+                    # saving the file to the disk.
```
In general, I allowed code (and psuedo-code and shell commands) that was inlined in a paragraph to be wrapped as ordinary text. (There were several cases that already did that, and the autofixer merely changed the wrap point.)

medmunds commented on django/db/backends/base/features.py:
```diff
@@ -195,12 +195,13 @@ class BaseDatabaseFeatures:
-    # supported by the Python driver
+    # Does the backend support 'pyformat' style ("... %(name)s ...", {'name':
+    # value}) parameter passing? Note this can be provided by the backend even
+    # if not supported by the Python driver
```
See previous comment. Code run into text is treated as text.

medmunds commented on django/db/backends/base/introspection.py:
```diff
@@ -168,8 +169,8 @@ def get_sequences(self, cursor, table_name, table_fields=()):
         """
-        Return a dictionary of {field_name: (field_name_other_table, other_table)}
-        representing all foreign keys in the given table.
+        Return a dictionary of {field_name: (field_name_other_table,
+        other_table)} representing all foreign keys in the given table.
```
I left this wrapped as text; there are several other examples of inline "code" wrapped as text prior to this change.

medmunds commented on django/dispatch/dispatcher.py:
```diff
@@ -29,7 +29,8 @@ class Signal:
         receivers:
-            [((id(receiver), id(sender)), ref(receiver), ref(sender), is_async)]
+            [((id(receiver), id(sender)), ref(receiver), ref(sender),
+            is_async)]
```
I'm not sure blackening this would be helpful. I left it alone

medmunds commented on django/template/smartif.py:
```diff
@@ -61,8 +61,8 @@ def eval(self, context):
-                # most likely to get exceptions for things like {% if foo in bar
-                # %} where 'bar' does not support 'in', so default to False
+                # most likely to get exceptions for things like {% if foo in
+                # bar %} where 'bar' does not support 'in', so default to False
```
This is a good example of pre-existing code treated (and wrapped) as text. I've allowed that pattern to remain elsewhere where introduced by the autofixer

medmunds commented on tests/cache/tests.py:
```diff
@@ -302,11 +304,10 @@ def custom_key_func(key, key_prefix, version):
+    # `base` is used to pull in the memcached config from the original
+    # settings, `exclude` is a set of cache names denoting which
+    # `_caches_setting_base` keys should be omitted. `params` are test specific
+    # overrides and `_caches_settings_base` is the base config for the tests.
```
I can't decide if this was meant to be a list (and the autofixer mangled it) or just running text

medmunds commented on tests/gis_tests/distapp/tests.py:
```diff
@@ -706,7 +721,8 @@ def test_length(self):
         #      ST_GeomFromText('<wkt>', 4326)
-        #      'SPHEROID["WGS 84",6378137,298.257223563, AUTHORITY["EPSG","7030"]]'
+        #      'SPHEROID["WGS 84",6378137,298.257223563,
+        #        AUTHORITY["EPSG","7030"]]'
```
I'm not sure this wrapped SQL string remains valid SQL.

medmunds commented on tests/gis_tests/test_spatialrefsys.py:
```diff
@@ -86,9 +87,10 @@ def test_retrieve(self):
+            # Some of the authority names are borked on Oracle, e.g.,
+            # SRID=32140.
+            #  also, Oracle Spatial seems to add extraneous info to fields,
+            #  hence the the testing with the 'startswith' flag.
```
I didn't know what to do with this. (Also "the the".)

medmunds commented on tests/introspection/tests.py:
```diff
@@ -319,10 +319,10 @@ def assertDetails(
+            #              PRIMARY KEY     UNIQUE CONSTRAINT    UNIQUE INDEX
+            # MySQL     pk=1 uniq=1 idx=1  pk=0 uniq=1 idx=1  pk=0 uniq=1 idx=1
+            # Postgres  pk=1 uniq=1 idx=0  pk=0 uniq=1 idx=0  pk=0 uniq=1 idx=1
+            # SQLite    pk=1 uniq=0 idx=0  pk=0 uniq=1 idx=0  pk=0 uniq=1 idx=1
```
I have abbreviated "PostgreSQL" as "Postgres" to make this table fit. (Another option was shrinking all the column gaps to a single space, but that made it hard to read.)

medmunds commented on django/contrib/syndication/views.py:
```diff
@@ -41,7 +41,8 @@ def __call__(self, request, *args, **kwargs):
-            # header so as ConditionalGetMiddleware is able to send 304 NOT MODIFIED
+            # header so as ConditionalGetMiddleware is able to send 304 NOT
+            # MODIFIED
             response.headers["Last-Modified"] = http_date(
```
"NOT MODIFIED" or "304 NOT MODIFIED" should maybe be kept together

medmunds reviewed (commented):
I'm done with edits; this is ready for review.

nessita reviewed (commented):
Commit 1 comments (nothing to do, but reviewing commit by commit is weird in GH and it seems I loose comments between each commit review?)

nessita commented:
buildbot, test on oracle.

medmunds commented:
Thanks for turning this around so quickly!

> I have gone thru each commit and thru each change. I made some further "editorial" fixes which I squashed in the first commit. Unclear if we should blame-ignore that one as well, will check with Sarah next week during out 1-1.

Those all lgtm. I would make you a co-author on that first commit. (But I don't want to push and interrupt the pending builds; perhaps the merger could update it?)

The argument for *not* including the first commit in blame-ignore is that I changed some wording and example syntax in ways that may have altered the original intent. (I tried not to, and you would have caught it if I had. But we probably wouldn't blame-ignore these changes if they were part of some other PR.)

> I'm so looking forward to not having to check line-lengths in code anymore! 🤩

Me too!

nessita commented:
I believe Jenkins CI has failed reporting status back to this PR. I can see in Jenkins that the run for this PR is fully green:
<img width="613" height="573" alt="image" src="https://github.com/user-attachments/assets/b27b7719-3ac0-41da-9fb6-b954fb0acb94" />

For the Oracle run, I believe that one is stuck. The run for this PR reports "Build has been executing for 7 hr 14 min on focal2" but the previous run in that job "Took 1 hr 51 min on focal2".

Given this CI issues, and considering tests are all green, I'll merge.

nessita commented:
Commits that will be git-blame ignored:
* 78298b51629e14c0e472898b635bc819d47b7f27
* 69a93a88edb56ba47f624dac7a21aacc47ea474f
